src/plugins/intel_cpu/src/memory_desc/cpu_memory_desc.h

```diff
@@ -44,7 +44,8 @@ enum class LayoutType : unsigned {
     nspc,      // general per channels format
     ncsp,      // general planar
     nCsp8c,    // general channels blocked by 8
-    nCsp16c    // general channels blocked by 16
+    nCsp16c,   // general channels blocked by 16
+    cabd
```

Comment:
Do we really need such a creator in the list of common creators?

Reply:
There 2 places using cabd creator: https://github.com/openvinotoolkit/openvino/blob/ad723b5e5f883f8a6e26657a9786461fbc54d691/src/plugins/intel_cpu/src/nodes/memory.cpp#L554
and:
https://github.com/openvinotoolkit/openvino/blob/ad723b5e5f883f8a6e26657a9786461fbc54d691/src/plugins/intel_cpu/src/nodes/scaled_attn.cpp#L790-L792
 if removed, is there any good location to place the common creation function?

Reply:
I refactored this part so that now we have an "ArbitaryOrderDescCreator" class, which simplifies instancing memory descriptors with any axis order. Also I removed `cabd` from the `LayoutType` enum, so that it doesn't break the initial `ncsp` description paradigm.